todo-app/src/App.js

```diff
@@ -1,13 +1,110 @@
-import React, { Component } from 'react';
+import React from 'react'
+import ToDo from './ToDo.js'
+import { ToDos } from './toDos-DB.js'
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      ToDos: ToDos,
+      newToDo: {
+        todoDesc:""
+      },
+      upatingTodoIndex:null
+    }
+  }
+  onChangeToDoStatus = (toDoIndex) => {
+    let updatedToDos = this.state.ToDos.map((toDo, index) => {
+      if (index === toDoIndex) {
+        return Object.assign({}, toDo, { isDone: !this.state.ToDos[index].isDone})
+      }
+      return toDo;
+    });
+    this.setState({
+      ToDos: updatedToDos
+    })
+  }
+  handleNewToDo = (event) => {
+    let updatedToDo = Object.assign({}, this.state.newToDo,
+      {
+        [event.target.name]: event.target.value
+      })
+    this.setState({
+      newToDo: updatedToDo
+    })
+  }
+  sendForm = (event) => {
+    event.preventDefault()
+    let updatedToDo = Object.assign({}, this.state.newToDo, {
+      date: new Date(),
+      isDone: false
+    })
+    let updatedToDos = [...this.state.ToDos, updatedToDo]
+    this.setState({
+      ToDos: updatedToDos,
+      newToDo: {
+        todoDesc: ""
+      }
+    })
+  }
+  editToDo = (toDoIndex) => {
+    this.setState({
+      upatingTodoIndex:this.state.upatingTodoIndex?null:toDoIndex+1
```

Comment:
This logic is a bit tricky... When we write code, we write so we can understand it in the future and our colleagues can understand it as well! you are trying to do everything at once in one line. 

How would you explain this line of code to someone? How can you improve it? Maybe cancel action of a todo should trigger another action?